scripts/ai2-internal/run_with_beaker.py

```diff
@@ -66,56 +62,63 @@ def main(param_file: str, extra_beaker_commands: List[str]):
             "--file-friendly-logging"
         ]
 
-    # TODO(michaels): add back in the env list.
-    # Presently this makes the Beaker UI unusably cluttered.
-    command = [
-            '/usr/local/bin/beaker',
-            'experiment',
-            'run',
-            '--result-path',
-            '/output',
-            "--source",
-            f"{config_dataset_id}:/config.json"] + env + extra_beaker_commands + [image] + allennlp_command
-    print(' '.join(command))
-    subprocess.run(command, check=True)
+    dataset_mounts = []
+    for source in (args.source if args.source else []):
+        datasetId, containerPath = source.split(":") + "\n"
+        dataset_mounts.append({
+            "datasetId": datasetId,
+            "containerPath": containerPath
+        })
+    for source in [f"{config_dataset_id}:/config.json"]:
+        datasetId, containerPath = source.split(":")
+        dataset_mounts.append({
+            "datasetId": datasetId,
+            "containerPath": containerPath
+        })
+
+    for var in (args.env if args.env else []):
+        key, value = var.split("=")
+        env[key] = value
+
+    config_spec = {
+        "description": args.desc,
+        "blueprint": blueprint,
+        "resultPath": "/output",
+        "args": allennlp_command,
+        "datasetMounts": dataset_mounts,
+        "requirements": {
+            "cpu": float(args.cpu),
+            "memory": args.memory,
+            "gpuCount": args.gpu_count if args.gpu_count else 1,
+        },
+        "env": env
+    }
+    config_task = {"spec": config_spec}
+
+    config = {
+        "tasks": [config_task]
+    }
+
+    with open(output_path, "w") as output:
+        output.write(yaml.dump(config))
+
+    print(f"Beaker spec written to {output_path}.  Launch your job with the following command")
+    print(f"")
+    print(f"    beaker experiment create --file {output_path}")
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
 
     parser.add_argument('param_file', type=str, help='The model configuration file.')
-    parser.add_argument('--name', type=str, help='A name for the experiment.')
+    parser.add_argument('output_path', type=str, help='The destination to write the experiment spec.')
```

Comment:
@matt-gardner makes a good point about convenience here.  I'd recommend a slightly different pattern here.  Drop `output_path` and instead echo the spec to STDOUT, so it can be piped into Beaker directly, like so:

```bash
./run_with_beaker.py <args...> | beaker experiment create --file=-
```

You can also switch on whether the spec is automatically run.  One possible mechanism would be to offer a `--dry-run` flag.  In this flag's absence, invoke Beaker directly. If `--dry-run` is present, then just echo the spec so the user can redirect to file and/or Beaker on their own.

Another option is to print the command you would invoke, and invoke this script with `exec $(run_with_beaker.py)`. The output might be in the following format:
```bash
cat << EOF | beaker experiment create --file=-
# Spec Goes Here...
EOF
```

Either of these options also implies retaining the `--name` flag.

Reply:
Funny, I just read this after I added a `--dry-run` flag ;-)

I actually personally think it's a good thing to have an extra step, but I'm happy to do what works best for other users (commit coming).